src/analytics.ts

```diff
@@ -140,13 +137,32 @@ export const logEvent = ({
         value = DEFAULT_EVENT_VALUE;
     }
 
-    const params = {
+    return {
         eventCategory: category,
         eventAction: action,
         eventLabel: formatLabel(label),
         eventValue: value,
         ...fieldsObject,
     };
+};
+
+export const logEvent = (event: TrackingEvent): Promise<void> => {
+    let {name, ...params} = event;
+    if (!name) {
+        // If the event doesn't have a name, it's a legacy analytics event
+        if (!event.category || !event.action) {
+            console.warn(
+                'LegacyAnalyticsEvent should have "category" and "action"',
+                {
+                    category: event.category,
+                    action: event.action,
+                },
+            );
+            return Promise.resolve();
+        }
+        params = getLegacyAnalyticsEventParams(event as LegacyAnalyticsEvent);
+        name = event.category;
+    }
```

Comment:
This is the legacy part.
If the event has a name, then we assume it's using the new google anlaytics 4 / firebase format and we don't transform the event, just forward it to the native app